### DISCUSSION TITLE
understanding mapping/demapping optimization

### DISCUSSION THREAD

**Discussion by nbecker** [NVlabs/sionna#159] [General]
I've been working with the example autoencoder.  I've modified it to take an initial constellation as input (not QAM).
The initial constellation is a 32-apsk constellation composed of 3 rings.  The mapper and demapper are optimized to minimize BCE.

In the attached pictures, the 1st picture shows the inital constellation.  The 2nd picture shows the mapper constellation after some training.  The 3rd picture shows how the points were modified by training.

I don't understand what this training is doing.   Intuitively, it seems non-optimal to move some of the points so close together.  Any idea what's going on?

[autoencoder_sspa_2sps_3_32apsk_linear_const.pdf](https://github.com/NVlabs/sionna/files/11884561/autoencoder_sspa_2sps_3_32apsk_linear_const.pdf)

**Comment by SebastianCa** [comment]
Hi,

the behavior you are observing is indeed expected and well-known. Fig. 4 in [1] shows similar results for different SNR regions.

From a conceptual standpoint, we aim to maximize the achievable rate R (i.e., the average “bitwise mutual information” of the individual bit channels) and not the BER itself. It's worth noting that our goal here isn't to maximize the Euclidean distance across all points within the constellation. Intuitively, you would also avoid higher order modulation schemes for very low SNR regions. The transmitter simply learns to cluster points which only differ in one bit which effectively removes a few constellation points from the alphabet (as the receiver cannot distinguish them). However, the practical gains are typically small and require a strong FEC decoder to ensure low-error rates.

A more dynamic illustration of this concept can be viewed in an animation [2] which demonstrates the decision regions involved during the training.

I hope this explanation helps.

[1] https://arxiv.org/pdf/1911.13055.pdf
[2] https://www.inue.uni-stuttgart.de/img/videos/bit_ae_k4n2_v2_1.mp4
